### PR TITLE
Improve Design documentation

### PR DESCRIPTION
**Pull request by TooAngel**

### CODE REVIEW

File: doc/Design.md [hunk 1]
```diff
@@ -32,8 +32,22 @@ queue is used.
    - External room: Builds container, fills container, calls `carry` to get
    the energy.
  - `reserver` reservs an external controller and calls `sourcer`.
- - `carry` gets energy from the target container and fills structures and
- storage on the way back. If there is a creep in front the energy is transfered.
+ - `carry`
+    A carry transports energy from the target to the storage in the base (From sources, energy piles or storages in other rooms)
+    They use fixed precalculated paths
+    While moving they try to transfer energy to the next creep or if in base to other structures (tower, link, extension)
```
Comment (samogot) [inline doc/Design.md]: link, really? my carries never transfers to links

File: doc/Design.md [hunk 2]
```diff
@@ -32,8 +32,22 @@ queue is used.
...
+    So carries move forward if the have 'no' energy, backwards if the 'have' energy
```
Comment (samogot) [inline doc/Design.md]: the**y** x2

**Comment by democratic-collaboration** [comment]
A new review, yeah.

    Votes: 228/407
    Coefficient: 0.560196560197
    Merging in 4 days 8 hours
    Age 0 days 1 hours

File: doc/Design.md (hunk 1 above)
Comment (TooAngel) [inline doc/Design.md]: It should transfer energy to the link at the exits, but not the ones near sourcers or at the storage.

Comment (samogot) [inline doc/Design.md]: So, that's RCL8 only feature? Why not to transfer into links near sources, if carry pass near them?

Comment (TooAngel) [inline doc/Design.md]: the links capacity is (nearly) fully used by the sourcer, if carries use them, too, energy would drop on the ground at some point

File: doc/Design.md (hunk 2 above)
Comment (TooAngel) [inline doc/Design.md]: thx, fixed

File: doc/Design.md (hunk 1 above)
Comment (samogot) [inline doc/Design.md]: really? sourcers perfomance is 10 energy per tick, link performance is 800/20=40 energy per tick. Isn't it?

Comment (samogot) [inline doc/Design.md]: oh, i've checked docs. cooldown 20 is only in my case, it depends on liner distance. the worst case is 800/50=16 ept

Comment (TooAngel) [inline doc/Design.md]: Well, the link handling can surely be improved. I took the simplest low CPU approach, round robin over all links in the given time.

**Pull request merged**